packages/hooks/src/useEventListener.ts

```diff
@@ -38,22 +38,19 @@ export default function useEventListener<
   }, [eventCallback]);
 
   useEffect(() => {
-    if (!enabled) {
-      return;
-    }
+    const callback = (e: GlobalEventHandlersEventMap[Type]) => {
```

Comment:
Can we add a test to cover what bug this uncovered? I still don't fully understand it but I think if we had a test that covers this situation would help me.